crates/router/src/connector/trustpay/transformers.rs

```diff
@@ -0,0 +1,964 @@
+        api_models::payments::BankRedirectData::Ideal { .. } => TrustpayPaymentMethod::IDeal,
+        api_models::payments::BankRedirectData::Sofort { .. } => TrustpayPaymentMethod::Sofort,
+    }
+}
+
+fn get_mandatory_fields(
+    item: &types::PaymentsAuthorizeRouterData,
+) -> Result<TrustpayMandatoryParams, error_stack::Report<errors::ConnectorError>> {
+    let billing_address = item
+        .address
+        .billing
+        .clone()
+        .unwrap_or_default()
+        .address
+        .unwrap_or_default();
+    Ok(TrustpayMandatoryParams {
+        billing_city: billing_address
+            .city
+            .ok_or(errors::ConnectorError::MissingRequiredField {
+                field_name: "billing.address.city",
+            })?,
+        billing_country: billing_address.country.ok_or(
+            errors::ConnectorError::MissingRequiredField {
+                field_name: "billing.address.country",
+            },
+        )?,
+        billing_street1: billing_address.line1.ok_or(
+            errors::ConnectorError::MissingRequiredField {
+                field_name: "billing.address.line1",
+            },
+        )?,
+        billing_postcode: billing_address.zip.ok_or(
+            errors::ConnectorError::MissingRequiredField {
+                field_name: "billing.address.postcode",
+            },
+        )?,
+    })
+}
+
+impl TryFrom<&types::PaymentsAuthorizeRouterData> for TrustpayPaymentsRequest {
+    type Error = error_stack::Report<errors::ConnectorError>;
+    fn try_from(item: &types::PaymentsAuthorizeRouterData) -> Result<Self, Self::Error> {
+        let default_browser_info = BrowserInformation {
+            color_depth: 24,
+            java_enabled: false,
+            java_script_enabled: true,
+            language: "en-US".to_string(),
+            screen_height: 1080,
+            screen_width: 1920,
+            time_zone: 3600,
+            accept_header: "*".to_string(),
+            user_agent: "none".to_string(),
+            ip_address: None,
+        };
+        let params = get_mandatory_fields(item)?;
+        let amount = utils::to_currency_base_unit(item.request.amount, item.request.currency)?
+            .parse::<f64>()
+            .ok()
+            .ok_or_else(|| errors::ConnectorError::RequestEncodingFailed)?;
+        Ok(match item.request.payment_method_data {
+            api::PaymentMethodData::Card(ref ccard) => Ok(Self {
+                cards_payment_request: Some(PaymentRequestCards {
+                    amount: format!("{:.2}", amount),
+                    currency: item.request.currency.to_string(),
+                    pan: ccard.card_number.peek().clone(),
+                    cvv: ccard.card_cvc.peek().clone(),
+                    expiry_date: format!(
+                        "{}/{}",
+                        ccard.card_exp_month.peek().clone(),
+                        ccard.card_exp_year.peek().clone()
+                    ),
```

Comment:
use `secret` field for pan, cvv and expiry date. During `serialization` this will be exposed and sent to the gateway

Reply:
Yes, moving this to utils